go/networkqualityd.go

```diff
@@ -14,26 +15,42 @@ import (
 	"net/http"
 	"os"
 	"os/signal"
+	"strings"
+	"syscall"
+
 	"strconv"
 	"sync"
-	"time"
 
 	// Do *not* remove this import. Per https://pkg.go.dev/net/http/pprof:
 	// The package is typically only imported for the side effect of registering
 	// its HTTP handlers. The handled paths all begin with /debug/pprof/.
-	_ "net/http/pprof"
 	// See -debug for how we use it.
+	_ "net/http/pprof"
+
+	"github.com/lucas-clemente/quic-go"
+	"github.com/lucas-clemente/quic-go/http3"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+	"golang.org/x/sys/unix"
 )
 
 var (
-	configPort = flag.Int("config-port", 4043, "The port to listen on for generating config responses")
-	publicPort = flag.Int("public-port", 4043, "The port to listen on for measurement accesses")
+	insecurePublicPort = flag.Int("insecure-public-port", 4080, "The port to listen on for measurement accesses")
+	publicPort         = flag.Int("public-port", 4043, "The port to listen on for measurement accesses")
+
 	listenAddr = flag.String("listen-addr", "localhost", "address to bind to")
 
-	debug = flag.Bool("debug", false, "enable debug mode")
+	announce               = flag.Bool("announce", false, "announce this server using DNS-SD")
+	debug                  = flag.Bool("debug", false, "enable debug mode")
+	enableCORS             = flag.Bool("enable-cors", false, "enable CORS headers")
+	enableH2C              = flag.Bool("enable-h2c", false, "enable h2c (non-TLS http/2 prior knowledge) mode")
+	enableHTTP2            = flag.Bool("enable-http2", true, "enable HTTP/2")
+	enableHTTP3            = flag.Bool("enable-http3", false, "enable HTTP/3")
+	enableInsecureListener = flag.Bool("enable-insecure-listener", false, "enable insecure listener")
```

Comment:
Should this be implied if the user gives `insecure-public-port`?

Reply:
done.